ee/clickhouse/queries/clickhouse_stickiness.py

```diff
@@ -56,48 +66,79 @@ def stickiness(self, entity: Entity, filter: StickinessFilter, team_id: int) ->
+        return retrieve_stickiness_people(filter, team)
 
-        parsed_date_from, parsed_date_to, _ = parse_timestamps(filter=filter, team_id=team.pk)
-        prop_filters, prop_filter_params = parse_prop_clauses(filter.properties, team.pk)
-        entity_sql, entity_params = self._format_entity_filter(entity=filter.target_entity)
-        trunc_func = get_trunc_func_ch(filter.interval)
-
-        params: Dict = {
-            "team_id": team.pk,
-            **prop_filter_params,
-            "stickiness_day": filter.selected_interval,
-            **entity_params,
-            "offset": filter.offset,
-        }
-
-        content_sql = STICKINESS_PEOPLE_SQL.format(
-            entity_filter=entity_sql,
-            parsed_date_from=parsed_date_from,
-            parsed_date_to=parsed_date_to,
-            filters=prop_filters,
-            trunc_func=trunc_func,
-        )
 
-        people = sync_execute(
-            PEOPLE_SQL.format(
-                content_sql=content_sql, query="", latest_person_sql=GET_LATEST_PERSON_SQL.format(query="")
+def _format_entity_filter(entity: Entity) -> Tuple[str, Dict]:
+    if entity.type == TREND_FILTER_TYPE_ACTIONS:
+        try:
+            action = Action.objects.get(pk=entity.id)
+            action_query, params = format_action_filter(action)
+            entity_filter = "AND {}".format(action_query)
+
+        except Action.DoesNotExist:
+            raise ValueError("This action does not exist")
+    else:
+        entity_filter = "AND event = %(event)s"
+        params = {"event": entity.id}
+
+    return entity_filter, params
+
+
+def _process_content_sql(filter: StickinessFilter, team: Team) -> Tuple[str, Dict[str, Any]]:
+    parsed_date_from, parsed_date_to, _ = parse_timestamps(filter=filter, team_id=team.pk)
+    prop_filters, prop_filter_params = parse_prop_clauses(filter.properties, team.pk)
+    entity_sql, entity_params = _format_entity_filter(entity=filter.target_entity)
+    trunc_func = get_trunc_func_ch(filter.interval)
+
+    params: Dict = {
+        "team_id": team.pk,
+        **prop_filter_params,
+        "stickiness_day": filter.selected_interval,
+        **entity_params,
+        "offset": filter.offset,
+    }
+
+    content_sql = STICKINESS_PEOPLE_SQL.format(
+        entity_filter=entity_sql,
+        parsed_date_from=parsed_date_from,
+        parsed_date_to=parsed_date_to,
+        filters=prop_filters,
+        trunc_func=trunc_func,
+    )
+    return content_sql, params
+
+
+def retrieve_stickiness_people(filter: StickinessFilter, team: Team) -> ReturnDict:
+
+    content_sql, params = _process_content_sql(filter, team)
+
+    people = sync_execute(
+        PEOPLE_SQL.format(content_sql=content_sql, query="", latest_person_sql=GET_LATEST_PERSON_SQL.format(query="")),
+        params,
+    )
+    return ClickhousePersonSerializer(people, many=True).data
+
+
+def insert_stickiness_people_into_cohort(cohort: Cohort, filter: StickinessFilter) -> None:
```

Comment:
I'm not sure these random functions in every query are that useful. I was already thinking that each query class should really have a `retrieve_people` function, so we can do the people thing in the frontend and the cohort thing here. what do you think?

Reply:
I agree with the `retrieve_people` func on all query classes. what do you mean by people thing on frontend and cohort thing here?